### PR TITLE
internal/docstore/dynamodocstore: add URL opener

### PR DESCRIPTION
**Pull request by shantuo**
Updates #1651

### CODE REVIEW

File: internal/docstore/dynamodocstore/dynamo.go [hunk 1]
```diff
@@ -12,22 +12,134 @@
...
 // limitations under the License.
 
+// Package dynamodocstore provides a docstore implementation backed by AWS
+// DynamoDB.
+// Use OpenCollection to construct a *docstore.Collection.
+//
+// URLs
+//
+// For docstore.OpenCollection, dynamodocstore registers for the scheme
+// "dynamodocstore". The default URL opener will use an AWS session with the
```
Comment (vangent) [inline internal/docstore/dynamodocstore/dynamo.go]: "dynamo" or "dynamodb" should be sufficient, here and elsewhere.

File: internal/docstore/dynamodocstore/dynamo.go [hunk 2]
```diff
@@ -12,22 +12,134 @@
...
+}
+
+// Scheme is the URL scheme dynamodocstore registers its URLOpener under on
+// docstore.DefaultMux.
+const Scheme = "dynamodocstore"
+
+// URLOpener opens dynamodocstore URLs like
+// "dynamodocstore://mytable?partition_key=partkey&sort_key=sortkey".
+//
+// The URL Host is used as the table name. See
```
Comment (vangent) [inline internal/docstore/dynamodocstore/dynamo.go]: Same question here as for Mongo: Is "foo/bar" a valid table name?

File: internal/docstore/dynamodocstore/dynamo.go [hunk 3]
```diff
@@ -12,22 +12,134 @@
...
+	overrideCfg, err := gcaws.ConfigFromURLParams(q)
+	if err != nil {
+		return nil, fmt.Errorf("open collection %s: %v", u, err)
+	}
+	configProvider.Configs = append(configProvider.Configs, overrideCfg)
+	db, err := Dial(configProvider)
+	if err != nil {
+		return nil, fmt.Errorf("open collection %s: %v", u, err)
+	}
+	return OpenCollection(db, tableName, partitionKey, sortKey), nil
```
Comment (vangent) [inline internal/docstore/dynamodocstore/dynamo.go]: OK for separate PR, but `OpenCollection` needs an `Options`, which should then be added here and to `URLOpener`.

File: internal/docstore/dynamodocstore/dynamo.go (hunk 1 above)
Comment (shantuo) [reply internal/docstore/dynamodocstore/dynamo.go]: Done.

File: internal/docstore/dynamodocstore/dynamo.go (hunk 2 above)
Comment (shantuo) [reply internal/docstore/dynamodocstore/dynamo.go]: No, I updated the link below directly to the naming rules.

File: internal/docstore/dynamodocstore/dynamo.go (hunk 3 above)
Comment (shantuo) [reply internal/docstore/dynamodocstore/dynamo.go]: Ack, filed #1877

**Pull request merged**